pipeline/http_post.go

```diff
@@ -14,48 +21,61 @@ package pipeline
 type HTTPPostNode struct {
 	chainnode
 
-	HTTPPostHandlers []*HTTPPostHandler `tick:"Endpoint"`
+	// tick:ignore
+	HTTPPostEndpoints []*HTTPPostEndpoint `tick:"Endpoint"`
 
 	// tick:ignore
-	Url string
+	urls []string
 }
 
 func newHTTPPostNode(wants EdgeType, urls ...string) *HTTPPostNode {
-	var url string
-	// TODO: What to do if len(urls) > 1?
-	if len(urls) != 0 {
-		url = urls[0]
-	}
 	return &HTTPPostNode{
 		chainnode: newBasicChainNode("http_post", wants, wants),
-		Url:       url,
+		urls:      urls,
 	}
 }
 
+// tick:ignore
+func (p *HTTPPostNode) Validate() error {
+	if len(p.urls) > 2 {
+		return fmt.Errorf("httpPost expects 0 or 1 arguments, got %v", len(p.urls))
+	}
+	return nil
+}
+
+func (p *HTTPPostNode) URL() (u string, err error) {
```

Comment:
While this is a nice clean up we can't do this as the public API of this struct is callable via TICKscript. We have to treat pipeline struct types as config structs with no behavior.